overlord/ifacestate/ifacestate.go

```diff
@@ -69,31 +69,22 @@ func findSymmetricAutoconnect(st *state.State, plugSnap, slotSnap string, autoCo
 	return false, nil
 }
 
-func checkConnectConflicts(st *state.State, change *state.Change, plugSnap, slotSnap string, autoConnectTask *state.Task) error {
-	if autoConnectTask == nil {
+func checkConnectConflicts(st *state.State, plugSnap, slotSnap string, auto bool) error {
```

Comment:
With the disconnect branch coming next we will need to think about some other name than 'auto' I think, as it will drive same conflict checks also for disconnect.  FWTW In the disconnect-hook PRs I renamed autoConnectTask to installedSnapTask